dockerfiles/Dockerfile

```diff
@@ -1,54 +1,113 @@
 
 ARG ARG_MIX_ENV=prod
+ARG ARG_NODE_MAJOR=18
+ARG ARG_WORKDIR=/app
 
-ENV MIX_ENV=${ARG_MIX_ENV} \
+ENV DEBIAN_FRONTEND=noninteractive \
+    WORKDIR="$ARG_WORKDIR" \
+    NODE_MAJOR=$ARG_NODE_MAJOR \
+    MIX_ENV=$ARG_MIX_ENV \
     MIX_HOME=/opt/mix
 
-WORKDIR /app
+WORKDIR $WORKDIR
 
-RUN apk --no-cache --update add alpine-sdk gmp-dev automake libtool inotify-tools autoconf python3 file gcompat libstdc++ curl ca-certificates gcompat nodejs npm git make
+COPY ./mix.exs ./mix.lock ./
+COPY ./apps ./apps
+COPY ./config ./config
+COPY ./rel ./rel
 
-# Cache elixir deps
-ADD mix.exs mix.lock ./
-ADD apps ./apps
-ADD config ./config
-ADD rel ./rel
+RUN set -euxo pipefail \
+    && apt update \
+    && apt -y install build-essential libtool ca-certificates curl gnupg git \
+    && mkdir -p /etc/apt/keyrings \
+    && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
+    && echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
+    && apt update \
+    && apt -y install nodejs \
+    && mix local.hex --force \
+    && mix do deps.get, local.rebar --force, deps.compile, compile \
+    && cd /app/apps/block_scout_web/assets/ && npm ci && npm run deploy \
+    && cd /app/apps/explorer/ && npm ci \
+    && cd /app && mix release blockscout
 
-RUN mix local.hex --force
-RUN mix do deps.get, local.rebar --force, deps.compile
-RUN mix compile
+##############################################################
+FROM ${ARG_IMAGE_NAME}:${ARG_IMAGE_VERSION}
 
-# Add blockscout npm deps
-RUN cd /app/apps/block_scout_web/assets/  \
-    && npm install npm@latest \
-    && npm install \
-    && npm run deploy \
-    && cd /app/apps/explorer/ \
-    && npm install \
-    && apk update \
-    && apk del --force-broken-world alpine-sdk gmp-dev automake libtool inotify-tools autoconf python3
+SHELL ["/bin/bash", "-c"]
 
-RUN export "CFLAGS=-I/usr/local/include -L/usr/local/lib" && cd deps/ex_secp256k1 && mix deps.get && mix compile
-RUN mix phx.digest
+ARG ARG_GOSU_VERSION=1.16
+ARG ARG_TINI_VERSION=v0.19.0
+ARG ARG_WORKDIR=/app
+ARG ARG_MIX_ENV=prod
 
-RUN mkdir -p /opt/release \
-  && mix release blockscout \
-  && mv _build/${MIX_ENV}/rel/blockscout /opt/release
+ENV DEBIAN_FRONTEND=noninteractive \
+    GOSU_VERSION="$ARG_GOSU_VERSION" \
+    TINI_VERSION="$ARG_TINI_VERSION" \
+    WORKDIR="$ARG_WORKDIR" \
+    MIX_ENV=${ARG_MIX_ENV}
 
-##############################################################
-FROM hexpm/elixir:1.14.5-erlang-25.3.2.4-alpine-3.18.2
+WORKDIR $WORKDIR
 
-ARG ARG_RELEASE_VERSION=5.2.1
-ENV RELEASE_VERSION=${ARG_RELEASE_VERSION}
+COPY --from=builder /app/_build/${MIX_ENV}/rel/blockscout ./
+COPY --from=builder /app/apps/explorer/node_modules ./node_modules
+COPY --from=builder /app/config/config_helper.exs ./config/config_helper.exs
+COPY ./dockerfiles/entrypoint.sh ./
```

Comment:
If I am not mistaken this will copy entrypoint to WORKDIR but you probably want it under `/usr/local/bin`